streamer/pulp/streamer/server.py

```diff
@@ -225,7 +226,23 @@ def close(self):
         """
         Forward the call to close the 'file' to the request.finish method.
         """
-        reactor.callFromThread(self.request.finish)
+        reactor.callFromThread(self.finish_wrapper)
+
+    def finish_wrapper(self):
+        """
+        Handles RuntimeErrors raised by calling ``finish`` on a request after
+        the connection is closed.
+
+        If finish is called after the client disconnects, a RuntimeError is
+        raised and Twisted logs the stack trace. Clients disconnecting before
+        Twisted gets around to calling ``finish`` is not uncommon. Ultimately
+        there should be a handler that stops streaming if the client disconnects,
+        but that is tricky with the current implementation's use of threads.
+        """
+        try:
+            self.request.finish()
+        except RuntimeError, e:
```

Comment:
use ``as`` instead of the comma for python3 compatibility.

https://www.python.org/dev/peps/pep-3110/